packages/services/src/boost-bonus.service.ts

```diff
@@ -517,6 +517,268 @@ export class BoostBonusService {
+   *
+   * **When to Remove:**
+   * - Competition's boosting window hasn't opened yet (boostStartDate > now)
+   * - New boostStartDate > boost.expiresAt (boost window starts after boost expires)
+   *
+   * **When to Keep:**
+   * - Boosting window already opened (boostStartDate < now) - user might have spent boost
+   * - Even if new boostStartDate is invalid, we keep boost for safety
+   *
+   * **Process:**
+   * 1. Find all boost_changes entries for this competition with meta.boostBonusId
+   * 2. Get unique boost IDs and fetch corresponding boost_bonus entries
+   * 3. For each boost where newBoostStartDate > boost.expiresAt:
+   *    - Check if previous boosting window already opened
+   *    - If not opened: Remove boost (decrease balance)
+   *    - If opened: Keep boost (could have been spent)
+   * 4. Use idempotency key: `cleanupInvalidBoost=${boostBonusId}&competition=${competitionId}`
+   *
+   * **Integration:**
+   * Called from CompetitionService.updateCompetition() when boostStartDate is updated.
+   *
+   * @param competitionId - The competition ID where boostStartDate changed
+   * @param newBoostStartDate - The new boostStartDate value
+   * @param oldBoostStartDate - The previous boostStartDate value (null if not set before)
+   * @param tx - Optional transaction (if not provided, creates new transaction)
+   * @returns Object containing lists of removed and kept boost IDs
+   * @throws Error if competition not found or cleanup fails
+   *
+   * @example
+   * ```typescript
+   * // Called when updating competition
+   * if (updates.boostStartDate !== undefined) {
+   *   await boostBonusService.cleanupInvalidBoostBonusesForCompetition(
+   *     competitionId,
+   *     updates.boostStartDate,
+   *     existingCompetition.boostStartDate,
+   *     tx
+   *   );
+   * }
+   * ```
+   */
+  async cleanupInvalidBoostBonusesForCompetition(
+    competitionId: string,
+    newBoostStartDate: Date,
+    oldBoostStartDate: Date | null,
+    tx?: Transaction,
+  ): Promise<{
+    removedBoostIds: string[];
+    keptBoostIds: string[];
+  }> {
+    const executeWithTx = async (transaction: Transaction) => {
+      const now = new Date();
+      const removedBoostIds: string[] = [];
+      const keptBoostIds: string[] = [];
+
+      this.#logger.info(
+        {
+          competitionId,
+          newBoostStartDate,
+          oldBoostStartDate,
+        },
+        "Starting cleanup of invalid bonus boosts for competition",
+      );
+
+      // Can't safely remove boosts if window already opened
+      if (
+        (oldBoostStartDate && oldBoostStartDate < now) ||
+        newBoostStartDate < now
+      ) {
+        this.#logger.info(
+          { competitionId, oldBoostStartDate, newBoostStartDate },
+          "Boosting window already opened - keeping all boosts for safety",
+        );
+        return {
+          removedBoostIds: [],
+          keptBoostIds: [],
```

Comment:
Technically, there _are_ boosts here right? We're just choosing not to enumerate them? This is a total nit, but should we then allow this to be nullable?